composable_kernel/include/utility/amd_buffer_addressing_v2.hpp

```diff
@@ -142,7 +183,13 @@ amd_buffer_load_impl_v2(int32x4_t src_wave_buffer_resource,
                         index_t src_wave_addr_offset)
 {
     static_assert((is_same<T, float>::value && (N == 1 || N == 2 || N == 4 || N == 8)) ||
-                      (is_same<T, int32_t>::value && (N == 1 || N == 2 || N == 4 || N == 8)),
+                      (is_same<T, half_t>::value && (N == 1 || N == 2 || N == 4)) ||
+                      (is_same<T, half2_t>::value && (N == 1)) ||
```

Comment:
why not just 

```
is_same<T, half_t>::value && (N == 1 || N == 2 || N == 4 || N = 8)
is_same<T, int32_t>::value && (N == 1 || N == 2 || N == 4)
```

Reply:
Since the input type is defined as typename vector_type<TInWei, InWeiVectorSize>::type. For fp16x4, the input type is half2_t.